app/workflow_executors/step_executors/impl/open_ai_next_js_code_generation_executor.go

```diff
@@ -161,27 +168,62 @@ func (openAiCodeGenerator OpenAiNextJsCodeGenerator) Execute(step steps.Generate
 	code, err := openAiCodeGenerator.GenerateCode(step, finalInstructionForGeneration, storyDir, apiKey)
 	if err != nil {
 		fmt.Println("____ERROR OCCURRED WHILE GENERATING CODE: ______", err)
-		settingsUrl := config.Get("app.url").(string) + "/settings"
-		err = openAiCodeGenerator.activityLogService.CreateActivityLog(
-			step.Execution.ID,
-			step.ExecutionStep.ID,
-			"INFO",
-			fmt.Sprintf("Action required: There's an issue with your LLM API Key. Ensure your API Key for %s is correct. <a href='%s' style='color:%s; text-decoration:%s;'>Settings</a>", constants.CLAUDE_3, settingsUrl, "blue", "underline"),
-		)
-		if err != nil {
-			fmt.Printf("Error creating activity log: %s\n", err.Error())
-			return err
-		}
-		//Update Execution Status and Story Status
-		if err = openAiCodeGenerator.storyService.UpdateStoryStatus(int(step.Story.ID), constants.InReviewLLMKeyNotFound); err != nil {
-			fmt.Printf("Error updating story status: %s\n", err.Error())
+		if err == types.ErrJsonParsingRetriesExceeded {
+			err = openAiCodeGenerator.slackAlert.SendAlert(
+				"Max retry limit reached while parsing JSON, error occurred while parsing the generated edit code",
```

Comment:
put alerts in all cases and with separate case of total max retry reached along with mention which model also in message